### ISSUE TITLE
Scan does not occur

### ISSUE THREAD

**Issue by MatthewGerber** [Estimote/Xamarin-Bindings#2]
In both the example Android project (in the bindings project) as well as my personal Xamarin Android project, the BT scan does not actually take place. BT is enabled, location permission is granted, and I get the following in the console after starting the proximity observer:

```
[ProximityAnalytics] Reporting start monitoring for 1bf17dbbd3810431f21644b965ebcd3d
[ProximitySDK] Sent usage metrics of one id.
```
That's the correct beacon identifier, so the Cloud credentials and JSON tag are working. However, as noted above, the scan does not occur, I don't get the usual console messages about BLE scanning, and no enter/exit events are reported by the observer. It just sits there without scanning.

This is on a Google Pixel running Android 8.0.

Any ideas?

**Comment by heypiotr** [comment]
Does the beacon have Estimote Monitoring enabled?

These bindings use our new Android Proximity SDK:
https://github.com/estimote/android-proximity-sdk

Instead of the soon-to-be-obsoleted "monolithic" Estimote Android SDK:
https://github.com/estimote/android-sdk

**Comment by MatthewGerber** [comment]
Thanks for the quick response! Yes, Estimote monitoring is enabled, and here are the location settings:

<img width="1329" alt="screen shot 2018-01-04 at 9 00 57 am" src="https://user-images.githubusercontent.com/4643066/34566776-d3ed013a-f12d-11e7-806e-d04c35337a7d.png">

What should I expect in terms of scanning frequency when using the sample Android app in the bindings project? I waited a few minutes with no enter/exit events, while the beacon was sitting next to me. Am I just not waiting long enough?

**Comment by heypiotr** [comment]
I've just realized this Android Xamarin bindings and example are missing the code to ask for runtime permission to access location data (`ACCESS_COARSE_LOCATION` or `ACCESS_FINE_LOCATION`).

We have a new helper library for that, `com.estimote:mustard`, which I'd need to bind and use in the example app. Or just port this native Android code: https://developer.android.com/training/permissions/requesting.html

I'll try to get to that Soon(ish) (:

**Comment by MatthewGerber** [comment]
Actually, in my own Xamarin Android project I am already requesting runtime location permissions prior to starting the Estimote proximity scan with a foreground service notification. Still, I get no scans or proximity events after waiting several minutes with the beacons sitting next to me. Is this what you would expect, or am I missing something else?

**Comment by heypiotr** [comment]
Oh, good to know, so it seems like runtime location permissions are not the main culprit. I'll have to do some more digging then, see if we have some debug mode available, try to run the [native Proximity SDK example](https://github.com/Estimote/Android-Proximity-SDK/tree/master/example/ProximityApp) to eliminate the prospect of there just being some problem with the SDK and not the binding … and see what other ideas I can come up with to pinpoint the issue ^_^

**Comment by MatthewGerber** [comment]
I just double-checked and confirmed that the Android testing app in this solution does not trigger a BLE scan, even when using low-latency mode and requesting location permissions in the manifest and at runtime. The app successfully gets the Estimote Beacon from the Cloud (and the beacon has monitoring enabled), and it outputs the following:

````
[ProximityAnalytics] Reporting start monitoring for 1bf17dbbd3810431f21644b965ebcd3d
[ProximitySDK] Sent usage metrics of one id.
````
After this, nothing happens. No BLE scan, no enter-handler events, etc. I haven't had a chance to run the native Android proximity app because I am Xamarin only right now.

Do you have any ideas about what might be going on with the bindings? I'm using a Google Pixel with Android 8.0.

**Comment by heypiotr** [comment]
I just updated the binding to alpha.6 and tweaked the example app to properly handle the location permissions. After that, the example app started working just fine on my Nexus 5X with Android 8.1.0. (It didn't work for me previously because there was a bug in the native alpha.5 SDK affecting Android 8.1.0.)

Can you fetch the latest version and see if it also helps your problem? You'll need to run `download_jars.sh` again.

**Comment by MatthewGerber** [comment]
I can confirm that the new version works, but only if _not_ using `WithScannerInForegroundService`. If I create a notification with the following code:
```
Notification notification = new Notification.Builder(Application.Context)
                                                        .SetContentTitle("Beacon scan")
                                                        .SetContentText("Scan is running...")
                                                        .Build();
```
and pass this notification to `WithScannerInForegroundService(notification)`, then the original problem returns, in that the following output is produced but no scans are started thereafter:

```
[ProximityAnalytics] Reporting start monitoring for 1bf17dbbd3810431f21644b965ebcd3d
[ProximitySDK] Sent usage metrics of one id.
```
Thoughts on how to fix this up?

**Comment by heypiotr** [comment]
I was able to easily reproduce this, and fix it in 66f67114b194ceb61146a871fbb1c0126ee32d84. 

Short story: take a look at the changes in the AndroidManifest.xml file (the `<service>` declaration), add it to your own app's manifest, and let me know if that helped (: 

Long story is ...

... when generating the bindings, Xamarin can only handle one AAR at the time, and the new Proximity SDK actually consists of multiple AARs—we have separate libraries for Cloud communication, low-level BLE scanning, etc.

What I ended up doing to work around that limitation is to only leave the high-level `proximity-sdk` library (which pulls the rest of the stuff via dependencies) as an AAR, and for the rest, I'm extracting the `classes` JARs from within the AARs, and using those instead. You can see this here:

[Estimote.Android.Proximity/Scripts/download_jars.sh#L11](https://github.com/Estimote/Xamarin-Bindings/blob/master/Estimote.Android.Proximity/Scripts/download_jars.sh#L11)

However, this means that the AndroidManifest.xml files from those AARs are not making it into the final manifest, and turns out one of the AARs (`scanning-sdk`) was not registering its internal scanning service because of that.

So the quick fix is to manually add this service to the app's AndroidManifest.xml:

https://github.com/Estimote/Xamarin-Bindings/commit/66f67114b194ceb61146a871fbb1c0126ee32d84#diff-31bd6ce5e52b12523bf7216e92595bb2R6

I'll also make an annotation about it in a README file—I'm working on a list of little things to keep in mind/remember about when using the bindings as they are here, for both iOS and Android, so I'll have that ready a bit later.

(There's another, potentially more elegant solution, to make the `download_jars.sh` script manually merge the AndroidManifest.xml files from all the AARs it ends up unpacking, into the "master" manifest from the `proximity-sdk` AAR. I'll have to consider that.)

**Comment by MatthewGerber** [comment]
Thanks so much for your work on this. Everything now works perfectly on Android. I have yet to test iOS.

I am quite impressed by the new proximity/location monitoring.
